From: Ben Hutchings <ben@decadent.org.uk>
Date: Fri, 12 Sep 2014 13:24:26 +0100
Subject: i2o: Disable I2O_EXT_ADAPTEC on 64-bit
Forwarded: no

The code it enables works uses 32-bit numbers for userland virtual
addresses:

In file included from /tmp/buildd/linux-3.16.2/drivers/message/i2o/config-osm.c:39:0:
/tmp/buildd/linux-3.16.2/drivers/message/i2o/i2o_config.c: In function 'i2o_cfg_passthru':
/tmp/buildd/linux-3.16.2/drivers/message/i2o/i2o_config.c:893:19: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
         (p->virt, (void __user *)sg[i].addr_bus,
                   ^
/tmp/buildd/linux-3.16.2/drivers/message/i2o/i2o_config.c:953:10: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
         ((void __user *)sg[j].addr_bus, sg_list[j].virt,
          ^

--- a/drivers/message/i2o/Kconfig
+++ b/drivers/message/i2o/Kconfig
@@ -37,6 +37,7 @@ config I2O_LCT_NOTIFY_ON_CHANGES
 
 config I2O_EXT_ADAPTEC
 	bool "Enable Adaptec extensions"
+	depends on !64BIT || BROKEN
 	default y
 	---help---
 	  Say Y for support of raidutils for Adaptec I2O controllers. You also
